### PR TITLE
Merged sl and extractor to one code base

### PR DESCRIPTION
**Pull request by hza**
to build extractor create `settings.properties` with content:

```
configuration=debug-variable-extractor
```

and run dvslBuild gradle task as usual.

Extractor uses minimal inner-tool (only 22 Kb) and does not have save/load features. JSON generation is done using ValueJsonSerializer class. JSON pretty formatting is done using jackson library.

screenshots:

ide

java

json

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -40,10 +40,19 @@ patchPluginXml {
     changeNotes """
       2.1.0 added extracting variable as JSON to the dialog <br/>
       2.0.0 added 'Copy Value as Java Code' dialog with options support <br/>
-      1.4.0 New: Android support. <br/>
```
Comment (hza) [inline build.gradle:43]: common  changeNotes for both

File: src/main/java/action/DebugVarAction.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import com.sun.jdi.Value;
+import org.jetbrains.annotations.NotNull;
+import util.debugger.ValueJsonSerializer;
+
+import java.awt.*;
+import java.awt.datatransfer.StringSelection;
+import java.util.HashSet;
+import java.util.Objects;
+
+public class DebugVarAction extends XDebuggerTreeActionBase {
```
Comment (hza) [inline src/main/java/action/DebugVarAction.java:21]: from Extractor

File: src/main/java/util/debugger/ValueJsonSerializer.java [hunk 3]
```diff
@@ -11,22 +11,23 @@
 import static util.debugger.ValueUtil.invokeMethod;
 
 public class ValueJsonSerializer {
-    public static final String JAVA_LANG_OBJECT = "java.lang.Object";
+    private static final String JAVA_LANG_OBJECT = "java.lang.Object";
+    private static final int MAX_DEPTH = 5;
```
Comment (hza) [inline src/main/java/util/debugger/ValueJsonSerializer.java:15]: adding MAX_DEPTH as constant for now

File: inner-tool/src/main/java/util/ObjectCodeGenerator.java [hunk 4]
```diff
@@ -488,4 +489,14 @@ private static Class<?> narrow(Class<?> clazz, Object object) {
 
         return finder;
     }
+
+    private static String escape(String raw) {
+        return raw.replace("\\", "\\\\")
+                .replace("\"", "\\\"")
+                .replace("\b", "\\b")
+                .replace("\f", "\\f")
+                .replace("\n", "\\n\"\n  + \"")
```
Comment (chocovon) [inline inner-tool/src/main/java/util/ObjectCodeGenerator.java:509]: It's different from ValueJsonSerializer#escape, what for?

Comment (hza) [reply inner-tool/src/main/java/util/ObjectCodeGenerator.java:509]: it is for better readability - these new lines with + in "hello world":

image

Impossible with JSON

**Pull request merged**